src/library/scala/beans/ScalaBeanInfo.scala

```diff
@@ -32,15 +31,9 @@ abstract class ScalaBeanInfo(clazz: java.lang.Class[_],
   override def getPropertyDescriptors() = pd
   override def getMethodDescriptors() = md
 
-  // override def getAdditionalBeanInfo() = Array(Introspector getBeanInfo clazz.getSuperclass)
-
   private def init() {
-    var i = 0
```

Comment:
I think it is probably _not_ fine because we would hope init is lightweight and this could make it quite a bit heavier (it adds three extra object allocations at least nominally: two to create the range and one for the closure).  Perhaps uses beans any more?  In any case, it shouldn't be changed without more justification that it doesn't matter performancewise (i.e. that this will not become a bottleneck in any reasonable scenario).  For a single property, I count four object allocations instead of one, for instance.